code/rounds/PreRound.cs

```diff
@@ -24,14 +26,32 @@ protected override void OnStart()
         {
             if (Host.IsServer)
             {
+                List<TTTAmmoRandom> randomAmmo = new();
+                List<TTTWeaponRandom> randomWeapons = new();
+
                 foreach (Entity entity in Entity.All)
                 {
                     if (entity is BaseCarriable carr)
                     {
                         carr.Delete();
                     }
+                    if (entity is TTTAmmo ammo)
+                    {
+                        ammo.Delete();
+                    }
+                    if (entity is TTTAmmoRandom rammo)
+                    {
+                        randomAmmo.Add(rammo); //Throws `Collection was Modified` if we activate here. Worth looking further into cleanup wise.
+                    }
+                    if (entity is TTTWeaponRandom rwep)
+                    {
+                        randomWeapons.Add(rwep); //See above comment.
+                    }
```

Comment:
Taking suggestions for speeding this process up if possible. May have questionable performance issues in the future with lots of entity processing.

Reply:
Hmm yea, what are you exactly doing here? You're looping through all Entities (Existing entities), adding them to a list, and then calling `Activate` for each item, that (in the Activate method) loops through all weapons again and spawns a random weapon in top of the already existing Entity? That's a lot of entities. Instead, these entities should get replaced, minimum. Also, you could store a list of available weapons globally, initializing it on Hotreload or Init.

Reply:
It iterates through all the random spawning entities and stores them in a list to be activated, which then spawns a new weapon entity.
The issue is that in my testing, activating it within the `foreach.. in Entity.All`, it'll throw that error `Collection was Modified` due to that new entity begin added.
Unfortunately I don't know a way to remedy this myself, which is why I left the comment. Might just make it an issue and leave it for another day.